app/src/reports/VariableReport.tsx

```diff
@@ -33,24 +33,18 @@ function VarGeoReport(props: {
     : ("diabetes_per_100k" as VariableId);
 
   const datasetStore = useDatasetStore();
-  const variableProvider = variableProviders[variableId];
-  const datasetIds = VariableProvider.getUniqueDatasetIds([variableProvider]);
+
+  const geoFilteredBreakdowns = Breakdowns.forFips(props.fips).andRace();
+  const allGeosBreakdowns = Breakdowns.byState().andRace();
+  const geoFilteredQuery = new VariableQuery(variableId, geoFilteredBreakdowns);
+  const allGeosQuery = new VariableQuery(variableId, allGeosBreakdowns);
+  const datasetIds = getDependentDatasets([variableId]);
 
   return (
-    <WithDatasets datasetIds={datasetIds}>
+    <WithVariables queries={[geoFilteredQuery, allGeosQuery]}>
       {() => {
-        let dataset = variableProvider.getData(
-          datasetStore.datasets,
-          Breakdowns.byState().andRace()
-        );
-
-        let tableDataset =
-          props.fips.code === USA_FIPS
-            ? variableProvider.getData(
-                datasetStore.datasets,
-                Breakdowns.national().andRace()
-              )
-            : dataset.filter((r) => r.state_fips === props.fips.code);
+        const mapDataset = datasetStore.getVariables(allGeosQuery);
```

Comment:
These could probably go directly in the component now instead of being pulled out as variables, right?

Reply:
Yup, done!